### DISCUSSION TITLE
Seata 2022-10-21 20:00 Monthly seminars

### DISCUSSION THREAD

**Discussion by funky-eyes** [apache/incubator-seata#4999] [Ideas]
The tentative topics are as follows：
1.https://github.com/seata/seata/issues/4957
2.https://github.com/seata/seata/issues/4956 https://github.com/seata/seata/issues/5007  https://github.com/seata/seata/issues/5005
3.https://github.com/seata/seata/issues/4997
4.https://github.com/seata/seata/pull/5004
5.summer code pr presentation （maybe）
Feel free to add relevant discussion topics by this Thursday

暂定议题如下：
1.https://github.com/seata/seata/issues/4957
- dbtype_sqltype spi扩展
- 放置1.6版本后再做优化

2.https://github.com/seata/seata/issues/4956
- 唯一索引为null的不作为条件
- 没有条件可用,不满足存在唯一索引,且主键自增情况走insert的逻辑
- insert into t_nounique values(3,3) on duplicate key update b = 5,这种sql需要根据matedata信息读到列的顺序去找到主键

3.https://github.com/seata/seata/issues/4997
- 当出现脏数据无法回滚时,应该上报二阶段状态为PhaseTwo_RollbackFailed_Unretryable,将全局事务改成RollbackFailed,避免出现issue中的情况
- 未来在控制台中突出不可重试的事务,提示用要求用户手动修订

4.https://github.com/seata/seata/pull/5004
优先1,如果能查到mode是什么,1,2都做,mode作为选择开关
- 保持当前pr group by pk做法
- groupby 查询列
- 代码层面去重 不推荐,可以做到list的去重

5.summer code pr presentation （可能）
欢迎在本周四之前补充相关讨论议题

**Comment by funky-eyes** [comment]
Personally, I think the https://github.com/seata/seata/pull/4907 should also be involved in the discussion

**Comment by pengten** [comment]
maybe #4527

**Comment by funky-eyes** [comment]
https://github.com/seata/seata/pull/5004  added the discussion
